### PR TITLE
dashboards: remove dashboards which are provided by node_exporter mixin

### LINKED ISSUE #234: Memory calculation needs a fix
**Why** - https://github.com/coreos/prometheus-operator/issues/1310#issue-320923699
https://github.com/coreos/prometheus-operator/issues/1310#issuecomment-387216413

**More Info** - https://github.com/coreos/prometheus-operator/issues/1310

### PR DESCRIPTION
**Pull request by paulfantom**
Remove dashboards provided by node_exporter mixin.

Closes https://github.com/coreos/prometheus-operator/issues/1310
Resolves #234

/cc @brancz @metalmatze

### CODE REVIEW

**Comment by brancz** [comment]
The removed file still needs to be removed from the dashboards.jsonnet. Then this will be green and lgtm.

cc @gouthamve @beorn7 @tomwilkie

**Comment by beorn7** [comment]
I was already working on a similar PR. Happy to take this one, though.

I was under the impression that a few recording rules can also be removed now. I'll double-check and add those here.

**Comment by beorn7** [comment]
Please also note that the node dashboard still has issues, which I have fixed in https://github.com/prometheus/node_exporter/pull/1450 . I could use a reviewer on that one. @metalmatze seems unresponsive.

**Comment by paulfantom** [comment]
I was also looking at removing some recording rules, but it appears they are also consumed by some alerts which are k8s specific. Didn't look very close though.

Removed dashboards from `dashboards.libsonnet`

**Comment by beorn7** [comment]
I have started checking for that. I'll finish it and let you know here about my results.

**Comment by beorn7** [comment]
All the following rules can go:
-  `:node_cpu_utilisation:avg1m`
-  `node:node_cpu_utilisation:avg1m`
-  `:node_cpu_saturation_load1:`
-  `node:node_cpu_saturation_load1:`
-  `:node_memory_utilisation:`
-  `node:node_memory_bytes_available:sum`
-  `node:node_memory_bytes_total:sum`
-  `node:node_memory_utilisation:ratio`
-  `:node_memory_swap_io_bytes:sum_rate`
-  `node:node_memory_utilisation:`
-  `node:node_memory_utilisation_2:`
-  `node:node_memory_swap_io_bytes:sum_rate`
-  `:node_disk_utilisation:avg_irate`
-  `:node_disk_saturation:avg_irate`
-  `node:node_disk_saturation:avg_irate`
-  `node:node_filesystem_usage:`
-  `node:node_filesystem_avail:`
-  `:node_net_utilisation:sum_irate`
-  `node:node_net_utilisation:sum_irate`
-  `:node_net_saturation:sum_irate`
-  `node:node_net_saturation:sum_irate`
-  `node:node_inodes_total:`
-  `node:node_inodes_free:`
-  `node:cluster_cpu_utilisation:ratio`
-  `node:cluster_memory_utilisation:ratio`

**Comment by beorn7** [comment]
@paulfantom Are you working on removing those rules, or do you prefer to let me do it in a separate PR?

**Comment by paulfantom** [comment]
I was just removing them :smile:

File: rules/rules.libsonnet [hunk 1]
```diff
@@ -243,102 +182,6 @@
...
-          {
-            // Disk utilisation (seconds spent, by rate() it's bound by 1 second)
-            record: ':node_disk_utilisation:avg_irate',
-            expr: |||
-              avg(irate(node_disk_io_time_seconds_total{%(nodeExporterSelector)s,%(diskDeviceSelector)s}[1m]))
-            ||| % $._config,
-          },
           {
             // Disk utilisation (seconds spent, by rate() it's bound by 1 second)
             record: 'node:node_disk_utilisation:avg_irate',
```
Comment (beorn7) [inline rules/rules.libsonnet]: Perhaps I missed it in my list, but this one can also go, IMHO.

Comment (paulfantom) [reply rules/rules.libsonnet]: removed

**Comment by paulfantom** [comment]
Removed `:node_memory_MemTotal_bytes:sum` and `node:node_disk_utilisation:avg_irate`

**Comment by beorn7** [comment]
Have you pushed an old commit? `node:node_disk_utilisation:avg_irate` and `:node_memory_MemTotal_bytes:sum` is still in there.

**Comment by paulfantom** [comment]
I don't what happened. https://github.com/kubernetes-monitoring/kubernetes-mixin/commit/aa3073ff78ab9f59e8374e4a81b2d9d1cb9ea9f2 Shouldn't have `node:node_disk_utilisation:avg_irate`, so I recreated that commit as https://github.com/kubernetes-monitoring/kubernetes-mixin/commit/b0ea7a444fbf9d051db9f288ce7730a4993c9526. Now it should be fine.

File: rules/rules.libsonnet [hunk 2]
```diff
@@ -168,284 +168,15 @@
...
-          {
-            record: ':node_memory_utilisation:',
-            expr: |||
-              1 -
-              sum(node_memory_MemFree_bytes{%(nodeExporterSelector)s} + node_memory_Cached_bytes{%(nodeExporterSelector)s} + node_memory_Buffers_bytes{%(nodeExporterSelector)s})
-              /
-              sum(node_memory_MemTotal_bytes{%(nodeExporterSelector)s})
-            ||| % $._config,
-          },
           // Add separate rules for Free & Total, so we can aggregate across clusters
```
Comment (beorn7) [inline rules/rules.libsonnet]: Nit: This comment is out of date now as there is no "total" anymore. :o)

**Comment by beorn7** [comment]
Thanks @paulfantom .

Now we need the authorities to merge this. :pray:

File: rules/rules.libsonnet (hunk 2 above)
Comment (paulfantom) [reply rules/rules.libsonnet]: fixed :)

**Pull request merged**